### PR TITLE
Setup rbac stuff

### PR DESCRIPTION
**Pull request by Ishankoradia**

### CODE REVIEW

File: ddpui/management/commands/role_based_access.py
Comment (fatchat) [inline ddpui/management/commands/role_based_access.py:1]: umm what

File: ddpui/core/orguserfunctions.py [hunk 1]
```diff
@@ -275,6 +277,86 @@ def invite_user(orguser: OrgUser, payload: InvitationSchema):
...
+        org=orguser.org, user__email__iexact=invited_email
+    ).exists():
+        return None, "user already has an account"
+
+    invited_role = Role.objects.filter(uuid=payload.invited_role_uuid).first()
+    if not invited_role:
+        return None, "Invalid role"
+
+    # user can only invite a role equal or lower to their role
+    if invited_role.level > orguser.new_role.level:
```
Comment (fatchat) [inline ddpui/core/orguserfunctions.py:298]: the proper check is whether the inviter has the correct permission to invite another user

File: ddpui/auth.py [hunk 2]
```diff
@@ -38,6 +46,32 @@ def authenticate(self, request, token):
...
+        @wraps(api_endpoint)
+        def wrapper(request, **kwargs):
+            # request will have set of permissions that are allowed
+            # check if permission_slug lies in this set
+            # throw error if nots
+            try:
+                if (
+                    request.permissions
+                    and len(
+                        set(permission_slugs).intersection(set(request.permissions))
```
Comment (fatchat) [inline ddpui/auth.py:60]: the request.permissions should be a subset of the permissions passed to the decorator

File: ddpui/core/orguserfunctions.py (hunk 1 above)
Comment (fatchat) [reply ddpui/core/orguserfunctions.py:298]: ok i see you're checking it in the middleware... in that case you can just remove this check

File: ddpui/management/commands/role_based_access.py
Comment (Ishankoradia) [reply ddpui/management/commands/role_based_access.py:1]: not clear what do you mean by that

File: ddpui/core/orguserfunctions.py (hunk 1 above)
Comment (Ishankoradia) [reply ddpui/core/orguserfunctions.py:298]: I think this should still be there.
 Even though the user has permissions to invite other users, an `account-manager` should not be able to invite a `super-user`.

File: ddpui/auth.py (hunk 2 above)
Comment (Ishankoradia) [reply ddpui/auth.py]: Agreed. made the changes

File: ddpui/core/orguserfunctions.py (hunk 1 above)
Comment (fatchat) [reply ddpui/core/orguserfunctions.py:339]: oh like that... hmm all right

**Review by fatchat — APPROVED** [review]

**Pull request merged**